node/external/nodeApiResolver.go

```diff
@@ -236,6 +242,11 @@ func (nar *nodeApiResolver) getInitialNodesPubKeysBytes(nodesInfo map[uint32][]n
 	return nodesInfoPubkeys
 }
 
+// GetGasConfigs return currently used gas schedule config
+func (nar *nodeApiResolver) GetGasConfigs() map[string]map[string]uint64 {
+	return nar.gasScheduleNotifier.LatestGasSchedule()
```

Comment:
the way LatestGasSchedule() is implemented makes this not concurrent safe.
Can you change also the LatestGasSchedule so that instead of returning the internal lastGasSchedule it creates a copy of it (under the mutex) and returns the copy instead?

Reply:
modified to return a copy of latestGasSchedule